.ci/azure/pypi.yml

```diff
@@ -0,0 +1,83 @@
+jobs:
+  - job: Build
+    pool:
+      vmImage: ubuntu-latest
+    steps:
+      # Checkout simpeg repo.
+      # Sync tags and disable shallow depth to get the SimPEG version.
+      - checkout: self
+        fetchDepth: 0
+        fetchTags: true
+        displayName: "Checkout repository (including tags)"
+
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: "3.9"
+        displayName: "Setup Python"
+
+      - bash: |
+          pip install build twine
+        displayName: "Install build dependencies"
+
+      - bash: |
+          # Change setuptools-scm local_scheme to "no-local-version" so the
+          # local part of the version isn't included, making the version string
+          # compatible with Test PyPI. Only do this when building for TestPyPI.
+          sed --in-place 's/node-and-date/no-local-version/g' pyproject.toml
+        condition: not(startsWith(variables['build.sourceBranch'], 'refs/tags/'))
+        displayName: "Configure local_scheme (except on release)"
+
+      - bash: |
+          python -m build --sdist .
```

Comment:
The biggest change I made after updating this branch once the `pyproject.toml` was merged was to change how we build the sdist. Instead of installing the build dependencies (and pin the versions here in the yml), I just install `build` and let it install the required build dependencies in isolation. This also solved an issue I saw when trying to push to TestPyPI, where the sdist didn't have the right simpeg version. I think it was related to incompatible versions of `setuptools` and `setuptools_scm`.